### ISSUE TITLE
Unable to run pycoco evaluation

### ISSUE THREAD

**Issue by furycurry** [ultralytics/ultralytics#18251]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have recently update to ultralytics 8.3.49 and tried running pycoco evaluation.  But all the metrices are showing 
```

> /usr/local/anaconda3/envs/john/bin/python3 /lab_202/john/project/ultralytics-8.3.49/ultralytics-main/pycocotools/pycoco.py 
> loading annotations into memory...
> Done (t=0.01s)
> creating index...
> index created!
> Loading and preparing results...
> DONE (t=0.00s)
> creating index...
> index created!
> Running per image evaluation...
> Evaluate annotation type *bbox*
> DONE (t=0.05s).
> Accumulating evaluation results...
> DONE (t=0.01s).
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
>  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
>  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000

```
what might be the reason for such?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @furycurry, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for guidance, including many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples where common workflows are documented.

If this is a 🐛 Bug Report, we kindly ask that you provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) so our team can better assist you in debugging the issue. This would include information like sample annotations, dataset structure, and all relevant usage details.

If this is a ❓ Question related to custom training or evaluation, please share as much supporting information as possible, such as training configuration details, evaluation commands, logs, and example outputs. Additionally, make sure you’ve reviewed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your pipeline.

### Troubleshooting and Upgrade

Ensure that you’ve upgraded to the latest version of `ultralytics` with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) met in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

### Additional Resources

Join the Ultralytics community! You can find engaging discussions or seek help across different channels:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- Deep discussions on [Discourse](https://community.ultralytics.com)
- Explore insights on our [Subreddit](https://reddit.com/r/Ultralytics)

### Environments

YOLO can be run in any of these verified environments with dependencies preinstalled:
- **Notebooks** with free GPU support: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### CI Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this CI badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, indicating expected functionality of YOLO across all [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

This is an automated response to provide immediate guidance 🔧. An Ultralytics engineer will review your issue and assist you further as soon as possible. Let us know if there’s anything else we can help you with in the meantime! 😊

**Comment by Y-T-G** [comment]
Is your category id starting from 1?

**Comment by furycurry** [comment]
No, its starting from 0

**Comment by Y-T-G** [comment]
Change it to start from 1 for your labels.

**Comment by furycurry** [comment]
How can I do that? you are telling to change the txt files to 1?

> `0 0.5817307692307693 0.30030959752321984 0.11538461538461539 0.30340557275541796`

**Comment by Y-T-G** [comment]
No. Your COCO label JSON file

**Comment by furycurry** [comment]
Category Id is starting from 1 in JSON file

**Comment by Y-T-G** [comment]
Is this predictions.json or the JSON with the labels?

**Comment by Y-T-G** [comment]
The latest version starts the category ID from 1 for predictions.json. So you have to change 0 to 1 in your labels json.

**Comment by furycurry** [comment]
Can you give a look if the format is okay now, if not do mention where is the error?
![{1BFFAEF1-1557-4DE4-BE12-777259D7B351}](https://github.com/user-attachments/assets/d0623f67-4458-4fe6-a9f5-22279a65da43)

**Comment by Y-T-G** [comment]
No. You have to replace all ``"category_id": 0` to ``"category_id": 1` in the file. Run find and replace or use `sed`.

**Comment by mldl4** [comment]
Hi!, sorry if I join to this conversation without permissions. I also have problems evaluating my model with pycocotools.
I trained a model that it has mid metrics. If I evaluate the model with ultralytics, the metrics are the expected, but if I got the preds (`save_json=true`) and I evaluate with pycocotools, my metrics show like that:

```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.004
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.007
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.005
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.004
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.004
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.031
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.005
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.012
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.013
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.008
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.016
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.053
```

As @Y-T-G commented, `category_id` and `image_id` is set to 1.

¿Any suggestions?

**Comment by Y-T-G** [comment]
What version of ultralytics are you using?

**Comment by mldl4** [comment]
I use ultralytics==8.3.39

**Comment by Y-T-G** [comment]
Can you try latest version?

**Comment by mldl4** [comment]
I updating the version and the results are much better. Although there are still differences. I will continue researching
